cloudinit/ssh_util.py

```diff
@@ -270,7 +270,6 @@ def setup_user_keys(keys, username, options=None):
     (auth_key_fn, auth_key_entries) = extract_authorized_keys(username)
     with util.SeLinuxGuard(ssh_dir, recursive=True):
         content = update_authorized_keys(auth_key_entries, key_entries)
-        util.ensure_dir(os.path.dirname(auth_key_fn), mode=0o700)
         util.write_file(auth_key_fn, content, mode=0o600)
```

Comment:
@ask0n won't the following `util.write_file(auth_key_fn, mode=0o600) also break in your scenario where " all files inside are owned by root because user shouldn’t be able to modify this files." ?

Reply:
@blackboxsw no, at least ssh access still will be functional and in the worst case user will be able to modify only his own keys.